packages/app/src/app/overmind/namespaces/workspace/actions.ts

```diff
@@ -444,3 +446,53 @@ export const openDefaultItem: Action = ({ state }) => {
 
   state.workspace.openedWorkspaceItem = defaultItem.id;
 };
+
+export const getDependencies: AsyncAction<string | void> = async (
+  { state, effects },
+  value
+) => {
+  state.workspace.loadingDependencySearch = true;
+  const searchResults = await effects.algoliaSearch.searchDependencies(value);
+
+  state.workspace.loadingDependencySearch = false;
+
+  if (!value) {
+    state.workspace.starterDependencies = [...searchResults];
+  }
+
+  state.workspace.dependencies = [...searchResults];
+};
+
+export const setDependencies: Action<Dependency[]> = (
+  { state },
+  newDependencies
+) => {
+  state.workspace.dependencies = [...newDependencies];
+};
+
+export const setSelectedDependencies: Action<Dependency> = (
+  { state },
+  dependency
+) => {
+  const oldDeps = state.workspace.selectedDependencies;
+  state.workspace.selectedDependencies = {
```

Comment:
A bit reduxy ;-)  You can do:

```ts
const selectedDependencies = state.workspace.selectedDependencies
if (selectedDependencies[dependency.objectID]) {
  delete selectedDependencies[dependency.objectID]
} else {
  selectedDependencies[dependency.objectID] = dependency
}

```

But no biggie, both works 😄

Reply:
10/10

This was copied from setState so that's why ahah